src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -208,9 +208,8 @@ where node.IsKind(SyntaxKind.IdentifierName)
 
             var bodyTokens = SyntaxUtilities.TryGetMethodDeclarationBody(node)?.DescendantTokens();
 
-            if (node.IsKind(SyntaxKind.ConstructorDeclaration))
+            if (node.IsKind(SyntaxKind.ConstructorDeclaration, out ConstructorDeclarationSyntax? ctor))
```

Comment:
Why is this marked with `?`?

Reply:
To satisfy NRT :)   if i don't, it talks about there being a nullability mismatch.

Reply:
Any chance `[MaybeNullWhen(false)] out TNode result` help here?

Reply:
Including the `?` is the _correct_ type for this call. If we had a feature like https://github.com/dotnet/csharplang/issues/727, then it would be reasonable to have states "non-null" and "not definitely assigned", but in the current language the states are "non-null" and "maybe null", which means it is not desirable for the `?` to be omitted.